contracts/strategies/BoostedJoeStrategyForLP.sol

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: MIT
+pragma experimental ABIEncoderV2;
+pragma solidity 0.7.3;
+
+import "../interfaces/IJoeChef.sol";
+import "../interfaces/IJoeVoterProxy.sol";
+import "../lib/SafeERC20.sol";
+import "./MasterChefStrategyForLP.sol";
+
+contract BoostedJoeStrategyForLP is MasterChefStrategyForLP {
+    using SafeERC20 for IERC20;
+
+    address public stakingContract;
+    IJoeVoterProxy public proxy;
+    address public swapPairRewardToken;
+
+    constructor(
+        string memory _name,
+        address _depositToken,
+        address _rewardToken,
+        address _nativeRewardToken,
+        SwapPairs memory _swapPairs,
+        address _stakingContract,
+        uint256 _pid,
+        address _voterProxy,
+        address _timelock,
+        StrategySettings memory _strategySettings
+    )
+        MasterChefStrategyForLP(
+            _name,
+            _depositToken,
+            _rewardToken,
+            _nativeRewardToken,
+            _swapPairs,
+            _timelock,
+            _pid,
+            _strategySettings
+        )
+    {
+        stakingContract = _stakingContract;
+        proxy = IJoeVoterProxy(_voterProxy);
+    }
+
+    function _depositMasterchef(uint256 _pid, uint256 _amount) internal override {
+        depositToken.safeTransfer(address(proxy), _amount);
+        proxy.deposit(_pid, stakingContract, address(depositToken), _amount);
+        proxy.distributeReward(_pid, stakingContract, address(extraToken));
```

Comment:
order here seems to be inverted, I'm sure there's a reason just could not grasp what it is.

Reply:
Line 46 automatically claims pending rewards to the voter and there is nothing we can do about it. So to get those rewards to the strategy they belong to we transfer them from the voter balance to the strategy balance and take our cut in line 47.